runtime/runtime_test.go

```diff
@@ -2531,169 +2537,125 @@ func TestRuntimeScriptReturnTypeNotReturnableError(t *testing.T) {
+				Location:  common.ScriptLocation{},
 			},
 		)
 
-		if expected == nil {
+		if test.invalid {
 			RequireError(t, err)
 
 			var subErr *InvalidScriptReturnTypeError
 			require.ErrorAs(t, err, &subErr)
 		} else {
 			require.NoError(t, err)
-			require.Equal(t, expected, actual)
+			require.Equal(t, test.expected, actual)
 		}
 	}
 
-	t.Run("function", func(t *testing.T) {
+	t.Run("interpreted function", func(t *testing.T) {
 
 		t.Parallel()
 
 		test(t,
-			`
-              pub fun main(): ((): Int) {
-                  return fun (): Int {
-                      return 0
+			testCase{
+				code: `
+                  pub fun main(): AnyStruct {
+                      return fun (): Int {
+                          return 0
+                      }
                   }
-              }
-            `,
-			nil,
-		)
-	})
-
-	t.Run("reference", func(t *testing.T) {
-
-		t.Parallel()
-
-		test(t,
-			`
-              pub fun main(): &Address {
-                  let a: Address = 0x1
-                  return &a as &Address
-              }
-            `,
-			cadence.Address{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1},
-		)
-	})
-
-	t.Run("recursive reference", func(t *testing.T) {
-
-		t.Parallel()
-
-		test(t,
-			`
-              pub fun main(): [&AnyStruct] {
-                  let refs: [&AnyStruct] = []
-                  refs.append(&refs as &AnyStruct)
-                  return refs
-              }
-            `,
-			cadence.NewArray([]cadence.Value{
-				cadence.NewArray([]cadence.Value{
-					nil,
-				}).WithType(cadence.VariableSizedArrayType{
-					ElementType: cadence.ReferenceType{
-						Type: cadence.AnyStructType{},
-					},
-				}),
-			}).WithType(cadence.VariableSizedArrayType{
-				ElementType: cadence.ReferenceType{
-					Type: cadence.AnyStructType{},
-				},
-			}),
-		)
-	})
-
-	t.Run("storage path", func(t *testing.T) {
-
-		t.Parallel()
-
-		test(t,
-			`
-              pub fun main(): StoragePath {
-                  return /storage/foo
-              }
-            `,
-			cadence.Path{
-				Domain:     "storage",
-				Identifier: "foo",
+                `,
+				expected: nil,
 			},
 		)
 	})
 
-	t.Run("public path", func(t *testing.T) {
+	t.Run("host function", func(t *testing.T) {
 
 		t.Parallel()
 
 		test(t,
-			`
-              pub fun main(): PublicPath {
-                  return /public/foo
-              }
-            `,
-			cadence.Path{
-				Domain:     "public",
-				Identifier: "foo",
+			testCase{
+				code: `
+                  pub fun main(): AnyStruct {
+                      return panic
+                  }
+                `,
+				expected: nil,
 			},
 		)
 	})
 
-	t.Run("private path", func(t *testing.T) {
+	t.Run("bound function", func(t *testing.T) {
 
 		t.Parallel()
 
 		test(t,
-			`
-              pub fun main(): PrivatePath {
-                  return /private/foo
-              }
-            `,
-			cadence.Path{
-				Domain:     "private",
-				Identifier: "foo",
```

Comment:
why are these removed?

Reply:
I noticed that the export of paths is already tested in `TestExportValue`, so removed the duplicates here.

Reply:
ah ok 👌